### ISSUE TITLE
Problem with Golems playing as iron teeth

### ISSUE THREAD

**Issue by Squat500** [hawkfalcon/Timberborn-Mods#12]
In the latest update, when playing as iron teeth, the game combines needs for both iron teeth and folktail golems. In this particular situation, placing any folktail assets to produce biofuel while playing as iron teeth crashes the game. Without biofuel, there is no way for golems to function after a short period of time. 

Disabling this mod eliminates the needs of folktail fuels while playing as iron teeth. I do not see any way to configure this in such a way that the needs of both factions do not both occur.

**Comment by Squat500** [comment]
![Uploading Broken.PNG…]()

**Comment by Squat500** [comment]
<img width="333" alt="Broken" src="https://user-images.githubusercontent.com/114884983/193478639-f87aaa27-db64-4e44-a5e8-1d769c8aad56.PNG">

**Comment by VerisZG** [comment]
Yes, and less funny is that building biofuel tank crashes my game.

**Comment by CDWe11s** [comment]
Similar issue when playing as folktails. Fortunately the biofuel assets don't crash the game and neither do the charging stations from ironteeth. However, it is really annoying that they need both fuel and energy to work... I was hoping to find a config to remove the golems as part of this mod... I guess that might be the next thing Hawkfalcon works on

**Comment by Cabal-2140** [comment]
Well with the latest updates of the game and mod, both factions are forced to the folktales version of the biofule refinery and it crashes the game as soon as it is build just 1%.

Only one error contained something like "refinery copy" witch even brought me to the problemic building after 5-6 crashes and so I testet only this mod and the building and it crashed again.

And because the mod got two updates since thees reports from october I guess it is not importent enought or known to be a thing again?

So I hope it gets fixed soon, have a nice holidays

**Comment by Just1nCasee** [comment]
More interesting that for me the error message doesn't cntain anything regarding to biofuel. Ths log has been created when the fuel refnery finished building and the game crashed (played as ironteeth facton). I think when you want to build a refinery, clicking on togle faction and placing the folk version of the buildng may doesn't lead to crash, but haven't tried. Still fixing the code would be nice.

InvalidOperationException: Sequence contains no matching element
System.Linq.Enumerable.Single[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] predicate) (at <5daa034a65ef4e92beb94b6ef26af36a>:0)
Timberborn.Stockpiles.GoodAmountTransformHeight.OnEnterFinishedState () (at <85a5cc0b1a654a7f85b40a6707ad1cea>:0)
Timberborn.ConstructibleSystem.Constructible.NotifyOnStateEntered () (at <70879c4cf8c8420ab037ce8973cb0f8c>:0)
Timberborn.ConstructibleSystem.Constructible.EnterConstructionState (Timberborn.ConstructibleSystem.ConstructionState constructionState) (at <70879c4cf8c8420ab037ce8973cb0f8c>:0)
Timberborn.ConstructibleSystem.Constructible.Finish () (at <70879c4cf8c8420ab037ce8973cb0f8c>:0)
Timberborn.ConstructionSites.ConstructionSite.FinishIfRequirementsMet () (at <da22411bbde44cfa8dbfd81310463f91>:0)
Timberborn.ConstructionSites.ConstructionSite.IncreaseBuildTime (System.Single hours) (at <da22411bbde44cfa8dbfd81310463f91>:0)
Timberborn.ConstructionSites.BuildExecutor.Tick (System.Single deltaTimeInHours) (at <da22411bbde44cfa8dbfd81310463f91>:0)
Timberborn.BehaviorSystem.BehaviorManager.TickRunningExecutor () (at <0e8e25b76d01493ab9031dd4b7bc969a>:0)
Timberborn.BehaviorSystem.BehaviorManager.Tick () (at <0e8e25b76d01493ab9031dd4b7bc969a>:0)
Timberborn.TickSystem.TickableComponent.StartAndTick () (at <906f400be9f14520940a09a425ac58cd>:0)
Timberborn.TickSystem.TickableEntity.TickTickableComponents () (at <906f400be9f14520940a09a425ac58cd>:0)
Timberborn.TickSystem.TickableEntity.Tick () (at <906f400be9f14520940a09a425ac58cd>:0)
Rethrow as Exception: Exception thrown while ticking entity 9bf35538-92d9-4e1e-a9fa-035c1088c3b7 'BeaverAdultZallah'
Timberborn.TickSystem.TickableEntity.Tick () (at <906f400be9f14520940a09a425ac58cd>:0)
Timberborn.TickSystem.TickableEntityBucket.TickAll () (at <906f400be9f14520940a09a425ac58cd>:0)
Timberborn.TickSystem.TickableBucketService.TickNextBucket () (at <906f400be9f14520940a09a425ac58cd>:0)
Timberborn.TickSystem.TickableBucketService.TickBuckets (System.Int32 numberOfBucketsToTick) (at <906f400be9f14520940a09a425ac58cd>:0)
Timberborn.TickSystem.Ticker.Update (System.Single deltaTimeInSeconds) (at <906f400be9f14520940a09a425ac58cd>:0)
Timberborn.TickSystem.TickerUpdater.Update () (at <906f400be9f14520940a09a425ac58cd>:0)

**Comment by Just1nCasee** [comment]
Also noticed that I am not able to store buifuel in tanks, even if I togle faction to folks.
